src/test/shared/settingsConfiguration.test.ts

```diff
@@ -89,63 +66,140 @@ describe('DefaultSettingsConfiguration', function () {
             // Setting exists but has wrong type:
             //
-            await settings.update('aws.telemetry', true, vscode.ConfigurationTarget.Global)
-            assert.deepStrictEqual(sut.getSetting<boolean>('aws.telemetry', 'function', { silent: 'yes' }), undefined)
-            assert.throws(function () {
-                sut.getSetting<boolean>('aws.telemetry', 'object', { silent: 'notfound' })
-            })
-            assert.throws(function () {
-                sut.getSetting<boolean>('aws.telemetry', 'string', { silent: 'no' })
-            })
-            assert.throws(function () {
-                assert.deepStrictEqual(
-                    sut.getSetting<boolean>('aws.telemetry', 'bigint', { silent: 'notfound' }),
-                    undefined
-                )
-            })
-            assert.throws(function () {
-                assert.deepStrictEqual(sut.getSetting<boolean>('aws.telemetry', 'symbol', {}), undefined)
-            })
-            assert.throws(function () {
-                assert.deepStrictEqual(sut.getSetting<boolean>('aws.telemetry', 'string'), undefined)
-            })
+            await settings.update(SETTING_KEY, true, SETTINGS_TARGET)
+            assert.throws(() => sut.getSetting(SETTING_KEY, String))
+            assert.throws(() => sut.getSetting(SETTING_KEY, Number))
+            assert.throws(() => sut.getSetting(SETTING_KEY, Object))
+            assert.throws(() => sut.getSetting(SETTING_KEY, Number))
         })
     })
 
     describe('writeSetting', async function () {
         scenarios.forEach(scenario => {
             it(scenario.desc, async () => {
-                await sut.writeSetting(SETTING_KEY, scenario.testValue, vscode.ConfigurationTarget.Global)
+                await sut.updateSetting(SETTING_KEY, scenario.testValue)
 
                 // Write tests need to retrieve vscode.WorkspaceConfiguration after writing the value
                 // because they seem to cache values.
-                const savedValue = vscode.workspace.getConfiguration(extensionSettingsPrefix).get(SETTING_KEY)
+                const savedValue = vscode.workspace.getConfiguration().get(SETTING_KEY)
 
                 assert.deepStrictEqual(savedValue, scenario.testValue)
             })
         })
     })
 
+    describe('fromPackage', function () {
+        const ProfileSettings = fromPackage('aws', { profile: String })
+        let settings: TestSettingsConfiguration
+        let instance: InstanceType<typeof ProfileSettings>
+
+        beforeEach(function () {
+            settings = new TestSettingsConfiguration()
+            instance = new ProfileSettings(settings)
+        })
+
+        it('throws if the setting does not exist', function () {
+            assert.throws(() => fromPackage('aws', { foo: Boolean }))
+        })
+
+        it('throws when the types do not match', function () {
+            assert.throws(() => instance.get('profile')) // `undefined` is not `string`
```

Comment:
some more tests for wrong types would add confidence :) this is the most fragile part of the vscode settings system (it will happily assign a string to a number at runtime).

Reply:
Sure, could add a few more. I have granular tests for `typeConstructors.ts`, just forgot to add them.